### ISSUE TITLE
Xenochimera feral-hallucinations happen in stomachs

### ISSUE THREAD

**Issue by Screemonster** [VOREStation/VOREStation#696]
#### Brief description of the issue
Xenochimera are not supposed to hallucinate when in sight of another person, and only get nag-messages. If they go feral in a stomach, or are swallowed while feral, it doesn't count them as being able to see someone.

#### What you expected to happen
To just get nag-messages as normal (or for messages to stop entirely)

#### What actually happened
Pun-pun started beating me up.

#### Steps to reproduce
Be xenochimera.
Go feral.
Get ate.

#### Additional info:
 - **Server Revision**: Server revision: release - 2016-10-22
6f3cb6bf7b33650a94068197e238f07f546e9567 
 - **Anything else you may wish to add** This will probably affect teshari too, once their alone-hallucinations are fixed. Would suggest skipping the whole proc if the player's location is a mob.

**Comment by Arokha** [comment]
This happens because the proc checks viewers(), aka, who can see Scree. Unfortunately every other fix adds slowdown, but there's really no avoiding it if you want it to work.

viewers() won't work when you can't see Scree.
view() returns every item, turf, etc as well as mobs, so is a much larger list.
hearers() is exactly the same as viewers().
hear() is what I want to use, but does not exist for whatever reason.

**Comment by Screemonster** [comment]
would something like... 
```
if (istype (scree.loc, /mob))
    skip all this shit
else
    do the thing
```
work, or would that be slow too?

**Comment by Arokha** [comment]
That'd be faster. The only thing I'd point out is that if you're in a macro's backpack, for example, you'll freak out. Or in a locker, even if you're in a locker with other people, you will, too. If that's fine, then go for it.

**Comment by Screemonster** [comment]
Well, the other would be to check if it's anything other than a /turf.

**Comment by Screemonster** [comment]
Fixed in #733
